### PR TITLE
Allow access to functions in inline assembly.

### LINKED ISSUE #513: Inline assembly: make function labels usable
If internal functions are accessed by name, the respective label should be pushed.

### PR DESCRIPTION
**Pull request by chriseth**
Fixes #513

### CODE REVIEW

**Comment by axic** [comment]
I was wondering if this could make it more confusing or not. Would it make sense prefixing those external method names so that it becomes easily visible it is an external method?

Also what happens in the case a function has the same name as an assembly instruction?

```
    function origin() private {
        assembly {
            origin()
        }
    }
```

**Comment by chriseth** [comment]
@axic assembly instructions of course take precendence. You can resolve the ambiguity by making an assignment to a solidity variable. Also, you cannot "call" the external functions in inline assembly, they behave more like jump labels.

**Comment by axic** [comment]
@chriseth so is this PR only adding support for jumping to the referenced method? That makes much more sense than allowing it to be a `CALL`.

File: docs/control-structures.rst [hunk 1]
```diff
@@ -452,15 +452,26 @@ Note that the order of arguments is reversed in functional-style as opposed to t
...
+Access to External Variables and Functions
+------------------------------------------
 
 Solidity variables and other identifiers can be accessed by simply using their name.
 For storage and memory variables, this will push the address and not the value onto the
 stack. Also note that non-struct and non-array storage variable addresses occupy two slots
 on the stack: One for the address and one for the byte offset inside the storage slot.
 In assignments (see below), we can even use local Solidity variables to assign to.
 
+Functions external to inline assembly can also be accessed: The assemble will
```
Comment (VoR0220) [inline docs/control-structures.rst]: "The assembly will"

**Comment by VoR0220** [comment]
Correct that typo and it's good to merge.

**Pull request merged**